apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -668,7 +668,9 @@ impl FetchDependencyGraph {
         //    keeping nodes separated when they have a different path in their parent
         //    allows to keep that "path in parent" more precisely,
         //    which is important for some case of @requires).
-        for existing_id in self.children_of(parent.parent_node_id) {
+        for existing_id in
+            FetchDependencyGraph::sorted_nodes(self.children_of(parent.parent_node_id))
+        {
```

Comment:
This will probably be slow when a fetch dependency node has very many children. Children are added one by one, so if this function is O(n), the complexity of adding many children is O(n^2). This is already a problem with `add_parent` and technically also with this loop, it will just be more pronounced if we also sort the children over and over.

I think it's okay if we do this here now, but we will have to improve the behaviour for nodes with many children in the near future.

Reply:
Realistically this will be at most in the order of hundreds of nodes (maybe couple thousand?) so don't think this will have much impact but yeah something to keep in mind.

Reply:
We already have a query where we spend multiple seconds in `add_parent` because of this 😄